### ISSUE TITLE
KDL layouts: Problematic node names

### ISSUE THREAD

**Issue by raphCode** [zellij-org/zellij#1818]
#1759
This is sort of an oddball problem, since users probably have to try to hit this problem, but I reported it nonetheless.
More pane node names can be defined than make sense.
We might display a more meaningful error that hints at quoting if that resolves the issue or forbid certain names altogether.

I have tested this only with pane template names but I am sure there can be more places where these problems apply.

### Empty string
```
layout {
    pane_template name="" {
        command "ls"
    }
    ""
}
```
works, but that seems strange somehow

### Numbers at the beginning
```
layout {
    pane_template name="0test" {
        command "ls"
    }
    0test
}
```
`Kdl Deserialization Error`
Quoting the second `0test` works.
Plain number without letters are also affected.

### Special characters
```
layout {
    pane_template name="hello(test)" {
        command "ls"
    }
    hello(test)
}
```
`Failed to deserialize KDL node.`

Basically the same case as with numbers, except now the special characters cause also trouble when they occur in the middle of the node name.
Quoting works again.
I haven't tested this exhaustedly, but I would guess there are a lot of special characters that may cause trouble.

**Comment by imsnif** [comment]
Btw, for the record `!@#$%^&*` is a valid template name and runs as expected ;P

**Comment by imsnif** [comment]
Fixed in https://github.com/zellij-org/zellij/pull/1831
